src/aiortc/contrib/media.py

```diff
@@ -913,6 +873,11 @@ async def __run_track(self, track, context):
                 for packet in context.stream.encode(display_frame):
                     self.__container.mux(packet)
 
+                if self.__save_dir is not None:
+                    display_array = display_frame.to_rgb().to_ndarray()
+                    np.save(os.path.join(self.__save_dir,
```

Comment:
if you have prediction enabled, isn't this gonna save frames twice? Is that desirable? and shouldn't this be under the same "save_predicted_frame" flag in which case you might want to call it something different "save_received_frames"

Reply:
I was not sure what the overall structure of possibilities will look like, meaning which percentage of the received frames will be predicted and which percentage will be real. I'll formulate it in a different way.

Reply:
Yes, I would avoid saving frames twice because that increases overheads drastically